### PR TITLE
add ghcup recipe

### PR DESCRIPTION
**Pull request by msarahan**
This is a tool for bootstrapping Haskell stuff.

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by msarahan** [comment]
@conda-forge/staged-recipes binary repack here. More info on ghcup and its importance in haskell on Apple Silicon here: https://www.reddit.com/r/haskell/comments/sifigb/how_to_use_stack_on_my_m1_mac_mini_with_apple/

File: recipes/ghcup/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  number: 0
+
+test:
+  commands:
+    - ghcup -h
+
+about:
+  home: https://gitlab.haskell.org/haskell/ghcup-hs
+  summary: An installer for the general purpose language Haskell
+  license: LGPL-3.0-only
```
Comment (isuruf) [inline recipes/ghcup/meta.yaml:32]: Is this enough? What about the dependencies?

Comment (msarahan) [reply recipes/ghcup/meta.yaml:32]: I'm not sure how to even go about finding these. What do you recommend?

Comment (isuruf) [reply recipes/ghcup/meta.yaml:37]: Ask upstream about it? They might know

Comment (ocefpaf) [reply recipes/ghcup/meta.yaml:41]: Looking at the repo I could not find any reference to any other license.

Comment (isuruf) [reply recipes/ghcup/meta.yaml:41]: Please don't merge with unresolved conversations. This package vendors in other haskell packages just like rust/go and we have mechanisms for those packages.

**Pull request merged**